### PR TITLE
test(replication): add admin port replication tests

### PR DESCRIPTION
**Pull request by kostasrim**
1. Addresses  #1470

### CODE REVIEW

File: tests/dragonfly/replication_test.py [hunk 1]
```diff
@@ -23,30 +23,57 @@
...
+        else df_local_factory.create(
+            port=BASE_PORT, admin_port=ADMIN_PORT, proactor_threads=t_master
+        )
+    )
+    replicas = (
+        [
+            df_local_factory.create(port=BASE_PORT + i + 1, proactor_threads=t)
+            for i, t in enumerate(t_replicas)
+        ]
+        if not from_admin_port
```
Comment (kostasrim) [inline tests/dragonfly/replication_test.py:60]: I am not particular fun of this but even if we extract it as a function, we still have to pass a bunch of arguments making it quite verbose. 

@adiholden is there a more pythonic way to do this and avoid the verbosity? 

1. If we extract the changes to a function, it's still amost the same + you have to pass a bunch of arguments on the caller side. 
2. If we copy paste the body of the test into a separate function, it's a redundant copying of 90% of the same test case.

File: tests/dragonfly/replication_test.py [hunk 2]
```diff
@@ -23,30 +23,57 @@
...
+            )
+            for i, t in enumerate(t_replicas)
+        ]
+    )
 
     # Start master
     master.start()
-    c_master = aioredis.Redis(port=master.port)
+    master_port = master.port if not from_admin_port else master.admin_port
+    c_master = (
```
Comment (adiholden) [inline tests/dragonfly/replication_test.py:72]: I dont think we need to set the replica/master client to use the admin port. Only the replicaiton should be over admin port i.e replicaof localhost master.admin_port

File: tests/dragonfly/replication_test.py [hunk 3]
```diff
@@ -23,30 +23,57 @@
...
-    df_local_factory, df_seeder_factory, t_master, t_replicas, seeder_config
+    df_local_factory, df_seeder_factory, t_master, t_replicas, seeder_config, from_admin_port
 ):
-    master = df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
-    replicas = [
-        df_local_factory.create(port=BASE_PORT + i + 1, proactor_threads=t)
-        for i, t in enumerate(t_replicas)
-    ]
+    master = (
+        df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
```
Comment (adiholden) [inline tests/dragonfly/replication_test.py:49]: you can just create always with admin port param, just if replication is from_admin_port use replica of master.admin_port

File: tests/dragonfly/replication_test.py [hunk 4]
```diff
@@ -23,30 +23,57 @@
...
-    (4, [8, 8], dict(keys=4_000, dbcount=4)),
-    (4, [1] * 8, dict(keys=500, dbcount=2)),
-    (1, [1], dict(keys=100, dbcount=2)),
+    (8, [8], dict(keys=10_000, dbcount=4), False),
+    (6, [6, 6, 6], dict(keys=4_000, dbcount=4), False),
+    (8, [2, 2, 2, 2], dict(keys=4_000, dbcount=4), False),
+    (4, [8, 8], dict(keys=4_000, dbcount=4), False),
+    (4, [1] * 8, dict(keys=500, dbcount=2), False),
+    (1, [1], dict(keys=100, dbcount=2), False),
+    (8, [8], dict(keys=10_000, dbcount=4), True),
```
[the remark this answers is not in the document]
Comment (kostasrim) [reply tests/dragonfly/replication_test.py:34]: I was thinking about the same on that, thanks for this comment, 100% agree

**Review by adiholden — APPROVED** [review]

**Pull request merged**